app/components/modal_component.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class ModalComponent < ViewComponent::Base
+  def initialize(id:, close_button: true)
+    @id = id
+    @close_button = close_button
+  end
+
+  private
+
+  def close_button_class
+    if namespace == "admin"
+      "red"
+    else
+      "primary"
+    end
```

Comment:
This should be the responsibility of the CSS. Do we have a `primary` class in admin already? That class name could be ambiguous as well. What if we created a new CSS class like `button_primary`?

But that may just be my opinion and it was in the code already. Ignoring.

Reply:
Actually, it is the original code and that's why I leave it here, but I agree with your comment: this should probably be the responsibility of the CSS.